### PR TITLE
Support more attributes for target redesign

### LINKED ISSUE #2956: Support readnone, readonly attributes
Support them to assist the target redesign change.

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR supports attributes: `unmangled`, `memory("read")`, `memory("none")`. 

This fixes #2956.

### CODE REVIEW

**Review by dbabokin — COMMENTED** [review]
Need to add description in the docs.

File: src/decl.cpp [hunk 1]
```diff
@@ -798,14 +801,25 @@ void Declarator::InitFromType(const Type *baseType, DeclSpecs *ds) {
             Warning(pos, "\"unmasked\" qualifier is redundant for exported "
                          "functions.");
 
+        if (isUnmangled) {
```
Comment (aneshlya) [inline src/decl.cpp:804]: A test covering different combinations of attributes/function qualifiers would be good.

Comment (nurmukhametov) [reply src/decl.cpp:804]: I have added a couple of tests.

**Comment by nurmukhametov** [comment]
> Need to add description in the docs.

I have added it for `unmangled`. About `memory`, I don't think we need to expose them to users.

**Review by aneshlya — APPROVED** [review]
Looks good!

**Pull request merged**